.github/workflows/sonar.yaml

```diff
@@ -0,0 +1,39 @@
+name: SonarCloud Analysis
+on:
+  push:
+    branches:
+      - main
+  pull_request_target:
+    types: [labeled]
+    branches:
+      - main
+    paths-ignore:
+      - 'LICENSE'
+      - '**/.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - '.github/**'
+      - 'kustomize/**'
+      - 'dev/**'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # Security due to https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+    if: github.repository == '5733d9e2be6485d52ffa08870cabdee0/sandbox' && contains(github.event.pull_request.labels.*.name, 'safe to test')
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Free disk space
+        uses: ./.github/actions/free-disk-space
+      - name: Java and Maven Setup
+        uses: ./.github/actions/java-maven-setup
+        with:
+          cache-key-prefix: ${{ runner.os }}
+      - name: Analyze with SonarCloud
+        uses: ./.github/actions/maven
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
+        with:
+          maven-command: -B verify sonar:sonar -Dmeasure-code-coverage -Dsonar.projectKey=5733d9e2be6485d52ffa08870cabdee0_sandbox -Dsonar.organization=5733d9e2be6485d52ffa08870cabdee0 -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=$SONAR_TOKEN
```

Comment:
@danielezonca well, we need to run tests, otherwise we will not get coverage, just code smells and bugs.

See results, where I replaced `verify` with `package -DskipTests` 

https://sonarcloud.io/summary/new_code?id=5733d9e2be6485d52ffa08870cabdee0_sandbox&branch=MGDOBR-282-tmp

Reply:
Ok I didn't notice the `push` event on `main` as trigger for this job too 🤔 
Is it possible to integrate the `sonar:sonar` goal in the `CI.yaml` file and keep this workflow only for `push` on `main`?
We are already reaching the max number of parallel jobs so duplicate build + test execution is not ideal 🤔

Reply:
@danielezonca that would be possible, if I change 'pull_request' CI trigger to 'pull_request_target' trigger. Because sonar job needs an access to repository secrets.

Reply:
I think this is ok, we already have multiple jobs using the `pull_request_target` event so it is fine for me to change `CI.yaml` to use it too

Reply:
changes pushed @danielezonca